benchmark/utils.py

```diff
@@ -0,0 +1,61 @@
+import os
+import subprocess
+import sys
+import time
+import torch
+
+model_dir = 'models/'
+install_file = 'install.py'
+hubconf_file = 'hubconf.py'
+
+
+def _install_deps(model_path):
+    if os.path.exists(os.path.join(model_path, install_file)):
+        subprocess.check_call([sys.executable, install_file], cwd=model_path)
+    else:
+        print('skip installing deps and preprocessing since no install.py is found in {}.'.format(model_path))
+
+
+def list_models():
+    return os.listdir(model_dir)
+
+
+def setup():
+    models = list_models()
+    for model in models:
+        _install_deps(model_dir + model)
+
+
+def run_models():
+    models = list_models()
+    for model in models:
+        cwd = os.getcwd()
+        model_path = os.path.join(cwd, model_dir, model)
+        sys.path.insert(0, model_path)
+        os.chdir(model_path)
```

Comment:
ideally we would put this logic in the list_models iterator and have `model` be the `Model` class for each backend.